eZ/Publish/Core/Persistence/Legacy/Content/Location/Gateway/DoctrineDatabase.php

```diff
@@ -1185,12 +1197,17 @@ public function loadTrashByLocation(int $locationId): array
         throw new NotFound('trash', $locationId);
     }
 
-    public function listTrashed(int $offset, ?int $limit, array $sort = null): array
-    {
+    public function listTrashed(
+        int $offset,
+        ?int $limit,
+        array $sort = null,
+        ?Criterion $criterion = null
+    ): array {
         $query = $this->connection->createQueryBuilder();
         $query
             ->select('*')
-            ->from('ezcontentobject_trash');
+            ->from(ContentGateway::CONTENT_ITEM_TABLE, 'c')
+            ->join('c', self::TRASH_TABLE, 't', 't.contentobject_id = c.id');
```

Comment:
One thing which came into my mind: we should adapt `>select('*')` to `->select('t.*')` . We don't need data from other tables then `self::TRASH_TABLE` :wink: